examples/widgets/text_cursor.py

```diff
@@ -0,0 +1,313 @@
+        ypos : float
+            The current y position of the cursor in data coordinates.
+            Important if *dataaxis* is set to 'y'.
+
+        Returns
+        -------
+        ret : {2D array-like, None} The coordinates which should be displayed.
+            *None* is the fallback value.
+        """
+
+        # Get plot line data
+        xdata = self.line.get_xdata()
+        ydata = self.line.get_ydata()
+
+        # The dataaxis attribute decides, in which axis we look up which cursor
+        # coordinate.
+        if self.dataaxis == 'x':
+            pos = xpos
+            data = xdata
+            lim = self.ax.get_xlim()
+        elif self.dataaxis == 'y':
+            pos = ypos
+            data = ydata
+            lim = self.ax.get_ylim()
+        else:
+            raise ValueError(
+                "The data axis specifier {} should be 'x' or 'y'"
+                .format(self.dataaxis))
+
+        # If position is valid
+        if pos is not None:
+            # And in valid plot data range
+            if pos >= lim[0] and pos <= lim[-1]:
+                # Convert given positon to numpy array,
+                # so numpy function can be used.
+                findme = np.array([pos])
+                # Find closest x value in sorted x vector.
+                # This is the code line,
+                # which requires the plotted data to be sorted.
+                index = np.searchsorted(data, findme)
+                # Return none, if this index is out of range.
+                if (index < 0) or (index >= len(data)):
+                    return None
+                # Return plot point as tuple.
+                return (xdata[index][0], ydata[index][0])
+
+        # Return none if there is no good related point for this x position.
+        return None
+
+    def clear(self, event):
+        """Overwirtten clear callback for cursor. Called right before \
+        displaying the figure."""
+
+        # The base class saves the clean background for blitting.
+        # Text and cursor are invisible,
+        # until the first mouse move event occurs.
+        super().clear(event)
+        if self.ignore(event):
+            return
+        self.text.set_visible(False)
+
+    def _update(self):
+        """
+        Overwritten method for eather blitting or drawing the widget canvas.
+
+        Passes call to base class if blitting is activated, only.
+        In other cases, one draw_idle call is enough, which is placed
+        explicitly in this class (see *onmove()*).
+        In that case, `~matplotlib.widgets.Cursor` is not supposed to draw
+        something using this method.
+        """
+
+        if self.useblit:
+            super()._update()
+
+
+fig = plt.figure(figsize=(8, 6))
+ax = fig.add_subplot(111, facecolor='#FFFFCC')
+
+# A linearly growing x vector.
+x = np.linspace(-5, 5, 1000)
+
+# A non biunique function. dataaxis='y' will cause trouble.
+y = (x ** 2)
```

Comment:
```suggestion
x = np.linspace(-5, 5, 1000)
y = x**2
```
is sufficient. No need to explain the obvious ("linearly growing x vector").

If you really want, add a `# sample data` comment on top.

I don't understand "dataaxis='y' will cause trouble.", but seems out of context here.

Reply:
You're right, the comment is indeed out of context.